fix: use --first-parent when loading Git hashes in GitDataSource (#26135)
cypress-io/cypress: cli/CHANGELOG.md

<!-- See the ../guides/writing-the-cypress-changelog.md for details on writing the changelog. -->
## 12.9.0

_Released 03/28/2023 (PENDING)_

> **Misc:**

Reviewer: This feels a bit more like a bug fix to me

Reviewer: or an enhacement to the debug experience. Not sure misc is the best section 🤷🏻‍♀️

Author: yeah that makes sense to me. I updated it to be under bugfixes

**Bugfixes:**

>  - Changed the way that we load Git hashes in order to exclude runs for non-relevant commits from the Debug page. Fixes [#26058](https://github.com/cypress-io/cypress/issues/26058).

Reviewer 2: nitpick: Should probably go under the Vite fix since that seems more significant, just my opinion though 🤷‍♂️

>  - Changed the way that we load Git hashes in order to exclude runs for non-relevant commits from the Debug page. Fixes [#26058](https://github.com/cypress-io/cypress/issues/26058).

Reviewer 2: ```suggestion
 - Changed the way that Git hashes are loaded so that non-relevant runs are excluded from the Debug page. Fixes [#26058](https://github.com/cypress-io/cypress/issues/26058).
```
